api/CatFacts.go

```diff
@@ -0,0 +1,48 @@
+package api
+
+import (
+	"net/http"
+	"errors"
+	"strings"
+)
+
+const catFactsBaseUrl = "https://catfact.ninja"
+
+const catFactEndpoint = catFactsBaseUrl + "/fact"
+const catFactFactKey = "fact"
+
+func GetCatFact() (string, error) {
+	resp, err := http.Get(catFactEndpoint)
+
+	if (err != nil) {
+		return "", err
+	}
+
+	defer resp.Body.Close()
+
+	if (resp.StatusCode != 200) {
+		return "", buildStatusError("Error recieved from cat facts endpoint", resp)
+	}
+
+	body, err := getBodyJson(resp)
+
+	if (err != nil) {
+		return "", err
+	}
+
+	if fact, exists := body[catFactFactKey]; exists {
+		return sanitizeCatFact(fact), nil
+	} else {
+		return "", errors.New("Could not parse fact out of response body!")
+	}
+}
+
+func GetCatFactAsync(channel chan StringResult) {
+	fact, err := GetCatFact()
+	channel <- StringResult{fact, err}
+}
+
+func sanitizeCatFact(fact string) string {
+	fact = strings.Replace(fact, "\"", "'", -1) // Remove quote marks
+	return strings.Replace(fact, "\\", "", -1) // Remove backslashes
+}
```

Comment:
This is more of an opinion piece: what would you think of doing this instead.
```
func removeQuotes(old string) (new string) {
    return strings.replace(old, "\"", "'", -1)
}

func removeBackslashes(old string) (new string) {
    return strings.Replace(old, "\\", "",  -1)
}

func sanitzeCatFact(fact string) string {
    return removeBackSlashes(removeQuotes(fact))
}
// or
func sanitzeCatFact(fact string) (sanitizedFact string) {
    sanitizedFact = removeQuotes(fact)
    return removeBackSlashes(sanitizedFact)
}
```

I consider it since you have a comment explaining `what` the two lines are doing and im under the opinion that breaking out little things for clarity *is* worth it, even if its more lines of code.

Also its very clear what is happening given the named return values `(new string)`

the different `sanitizeCatFact`s are up to you. I think idiomatic go would prefer the second version [Obvious code is important. What you can do in one line you should do in three](https://dave.cheney.net/practical-go/presentations/qcon-china.html#_choose_identifiers_for_clarity_not_brevity)